leptonai/cli/deployment.py

```diff
@@ -314,7 +316,8 @@ def create(
 
     # First, check whether the input is photon or container. We will prioritize using
     # photon if both are specified.
-    if photon is not None or photon_id is not None:
+    deployment_template = PhotonDeploymentTemplate()
```

Comment:
Hmm. I think this is not needed, as both the photon path and arbitrary container path explicitly sets deployment_template.

Reply:
I think the real issue is at line 343(old)/346(new) of the file:

deployment_template = photon.deployment_template

photon.deployment_template might be None, which causes the issue. You can change it to

deployment_template = photon.deployment_template or PhotonDeploymentTemplate()

could you also check if the above solution solves the bug you saw?

Reply:
Oh Yes, the issue was due to the use of only **if** and **elif** conditions. If photon_id, photon_name, and container info are not provided, the deployment_template would remain undeclared. Since I've added an else clause to provide a user hint and call sys.exit(1), this problem no longer occurs so we don't need do "deployment_template = PhotonDeploymentTemplate()" here.